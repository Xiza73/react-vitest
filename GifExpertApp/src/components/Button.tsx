export interface ButtonProps {
  text: string;
  onClick: (input?: string) => void;
}

export const Button: React.FC<ButtonProps> = ({ text, onClick }) => {
  return <button onClick={() => onClick()}>{text}</button>;
};
