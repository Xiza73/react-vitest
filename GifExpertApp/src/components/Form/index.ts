export * from './Browser';
export * from './Input';
