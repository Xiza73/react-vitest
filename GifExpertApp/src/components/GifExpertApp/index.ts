export * from './CategoriesContainer';
export * from './Gif';
export * from './GifsContainer';
