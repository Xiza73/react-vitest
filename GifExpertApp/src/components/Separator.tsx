export interface SeparatorProps {
  className?: string;
}

export const Separator: React.FC<SeparatorProps> = ({ className }) => (
  <hr className={`w-full my-4 ${className}`} />
);
