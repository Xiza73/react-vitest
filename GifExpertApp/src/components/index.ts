export * from './Form';
export * from './GifExpertApp';
export * from './Button';
