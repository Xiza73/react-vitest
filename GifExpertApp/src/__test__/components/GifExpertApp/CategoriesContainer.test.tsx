import { CategoriesContainer } from "@/components";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { act } from "react-dom/test-utils";

describe("Tests on <CategoriesContainer />", () => {
  const titles = ["One Punch", "Dragon Ball"];

  test("should match the snapshot", () => {
    const { container } = render(<CategoriesContainer titles={titles} />);

    expect(container).toMatchSnapshot();
  });

  test("should show the titles", () => {
    render(<CategoriesContainer titles={titles} />);

    titles.forEach((title) => {
      expect(screen.queryByText(new RegExp(title, "i"))).toBeTruthy();
    });
  });

  test("should show the gifs by category", async () => {
    render(<CategoriesContainer titles={titles} />);

    await waitFor(() => {
      expect(screen.queryByText(/loading.../i)).toBeNull();
    });

    expect(screen.getAllByRole("img").length).toBeGreaterThan(0);
    expect(screen.getAllByRole("img").length).toBe(titles.length * 8);
  });

  test("should show more gifs when Load More button is clicked", async () => {
    render(<CategoriesContainer titles={titles} />);

    await waitFor(() => {
      expect(screen.queryByText(/loading.../i)).toBeNull();
    });

    const btn = screen.queryAllByText(/load more/i)[0];

    act(() => {
      fireEvent.click(btn);
    });

    await waitFor(() => {
      expect(screen.queryByText(/loading.../i)).toBeNull();
    });

    expect(screen.getAllByRole("img").length).toBeGreaterThan(
      titles.length * 8
    );
  });
});
