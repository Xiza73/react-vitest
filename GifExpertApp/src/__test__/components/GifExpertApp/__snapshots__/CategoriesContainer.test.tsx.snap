// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests on <CategoriesContainer /> > should match the snapshot 1`] = `
<div>
  <ol>
    <li>
      <h2>
        One Punch
      </h2>
      <ol />
      <p>
        Loading...
      </p>
    </li>
    <li>
      <h2>
        Dragon Ball
      </h2>
      <ol />
      <p>
        Loading...
      </p>
    </li>
  </ol>
</div>
`;
